src/monodroid/jni/monodroid-glue.cc

```diff
@@ -1180,11 +1155,19 @@ MonodroidRuntime::monodroid_dlopen (const char *name, int flags, char **err, [[m
 	else
 		basename_part = (char*)name;
 
-	simple_pointer_guard<char[]> basename = utils.string_concat ("libaot-", basename_part);
-	h = androidSystem.load_dso_from_any_directories (basename, dl_flags);
+	constexpr char LIBAOT_PREFIX[] = "libaot-";
+	constexpr size_t LIBAOT_PREFIX_SIZE = sizeof(LIBAOT_PREFIX);
+
+	size_t base_len = strlen (basename_part);
+	size_t len = base_len + LIBAOT_PREFIX_SIZE; // includes the trailing \0
```

Comment:
This this also use the overflow-checking macro?

Reply:
The likelihood of a file path + some static string length exceeding `18446744073709551615` is quite low... :)